paparazzi/src/test/java/app/cash/paparazzi/HtmlReportWriterTest.kt

```diff
@@ -266,9 +267,106 @@ class HtmlReportWriterTest {
+
+      htmlReportWriter.use {
+        val horizontalFrameHandler = htmlReportWriter.newFrameHandler(
+          Snapshot(
+            name = "horizontal",
+            testName = TestName("app.cash.paparazzi", "SimilarViews", "horizontal"),
+            timestamp = Instant.parse("2019-03-20T10:27:43Z").toDate(),
+            tags = listOf("redesign")
+          ),
+          frameCount = 1,
+          fps = -1
+        )
+        horizontalFrameHandler.use {
+          horizontalFrameHandler.handle(convertedHorizontal)
+        }
+        val verticalFrameHandler = htmlReportWriter.newFrameHandler(
+          Snapshot(
+            name = "loading",
+            testName = TestName("app.cash.paparazzi", "SimilarViews", "vertical"),
+            timestamp = Instant.parse("2019-03-20T10:27:43Z").toDate(),
+            tags = listOf("redesign")
+          ),
+          frameCount = 1,
+          fps = -1
+        )
+        verticalFrameHandler.use {
+          verticalFrameHandler.handle(convertedVertical)
+        }
+        val squareFrameHandler = htmlReportWriter.newFrameHandler(
+          Snapshot(
+            name = "loading",
+            testName = TestName("app.cash.paparazzi", "SimilarViews", "square"),
+            timestamp = Instant.parse("2019-03-20T10:27:43Z").toDate(),
+            tags = listOf("redesign")
+          ),
+          frameCount = 1,
+          fps = -1
+        )
+        squareFrameHandler.use {
+          squareFrameHandler.handle(convertedSquare)
+        }
+      }
+      // Verify that there are 3 files in the reports/paparazzi/images file
+      // This should confirm that the snapshot images used in the test report are correct
+      val snapshotDir = File(snapshotRoot.root, "images").listFiles()!!
+      assertThat(snapshotDir.size).isEqualTo(3)
+
+      val firstSnapshot = ImageIO.read(snapshotDir[0])
+      val secondSnapshot = ImageIO.read(snapshotDir[1])
+      val thirdSnapshot = ImageIO.read(snapshotDir[2])
+      assertThat(firstSnapshot.pixelEqual(secondSnapshot)).isFalse()
+      assertThat(firstSnapshot.pixelEqual(thirdSnapshot)).isFalse()
+      assertThat(secondSnapshot.pixelEqual(thirdSnapshot)).isFalse()
+    } finally {
+      // reset record mode
+      System.setProperty("paparazzi.test.record", "false")
+    }
+  }
+
   private fun Instant.toDate() = Date(toEpochMilli())
 
   private fun File.lastModifiedTime(): FileTime {
     return Files.readAttributes(this.toPath(), BasicFileAttributes::class.java).lastModifiedTime()
   }
+
+  // Function that takes in an existing Buffered image and converts its type to one
+  // that ApngWriter can consume
+  private fun BufferedImage.convertImage() = BufferedImage(
+    width,
+    height,
+    BufferedImage.TYPE_INT_ARGB_PRE // Something ApngWriter can process
+  ).apply {
+    graphics.drawImage(this, 0, 0, null)
+  }
+
+  private fun BufferedImage.pixelEqual(other: BufferedImage): Boolean {
```

Comment:
Is there an existing comparator for this? I looked briefly but didn't want to block review.